cpp/src/cylon/indexing/slice.cpp

```diff
@@ -0,0 +1,170 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <memory>
+#include <algorithm>
+
+#include <cylon/table.hpp>
+#include <cylon/ctx/arrow_memory_pool_utils.hpp>
+#include <cylon/util/macros.hpp>
+#include <cylon/util/arrow_utils.hpp>
+#include <cylon/scalar.hpp>
+
+namespace cylon {
+
+static constexpr int64_t kZero = 0;
+
+/**
+ * Slice the part of table to create a single table
+ * @param table, offset and length
+ * @return new sliced table
+ */
+Status Slice(const std::shared_ptr<Table> &in, int64_t offset, int64_t length,
+             std::shared_ptr<Table> *out) {
+  const auto &ctx = in->GetContext();
+  const auto &in_table = in->get_table();
+
+  std::shared_ptr<arrow::Table> out_table;
+  if (!in->Empty()) {
+    out_table = in_table->Slice(offset, length);
+  } else {
+    out_table = in_table;
+  }
+  return Table::FromArrowTable(ctx, std::move(out_table), *out);
+}
```

Comment:
There is already a `out` pointer and also there is another `out_table` `shared_ptr` reference created to extract output. Plus the method takes in a Table `shared_ptr` and a pointer. This looks kind of redundant and not clean as per my understanding. May be the required change to the method is out of scope for this PR. 

@nirandaperera appreciate your feedback on this method and usage.